### ISSUE TITLE
User group is assigned proper Sprout Report permissions but can't run "Export".

### ISSUE THREAD

**Issue by benjaminkohl** [barrelstrength/craft-sprout-reports#83]
### Description

We created a client-level admin user group and gave them access to Sprout Reports in the control panel and checked all three of Sprout Report's permission settings. Yet when a user from this user group tries to run the "Export" feature on any of the reports, they get a Craft error screen that reads "User is not permitted to perform this action".

I looked in the Craft logs and this error is being thrown from line 474 of the ReportsController, which is the first line of actionExportReport().

```
$this->requirePermission($this->permissions['sproutReports-viewReports']);
```

### Additional info

- Craft version: 3.4.11
- PHP version: 7.4.4
- Database driver & version:  MySQL 5.5.5
- Plugins & versions: Sprout Reports 1.3.3

**Comment by ryanmasuga** [comment]
@BenParizek - Have you had a chance to look at this one? We upgraded a client from Craft 2 to Craft 3, and now the only thing they cannot do is run reports for themselves.

**Comment by BenParizek** [comment]
@ryanmasuga I'll be catching up on support this week and will let you know when I have more.

**Comment by ryanmasuga** [comment]
Touching base on this. Since upgrading the Craft 3, the client is asking us to run their reports every few days now because they can't do so themselves.

**Comment by ryanmasuga** [comment]
Second, different client wrote us today wondering why they cannot do report exports themselves. after getting an "Unauthorized - User is not permitted to perform this action" message.

**Comment by BenParizek** [comment]
@ryanmasuga I'm working through several updates right now and hope to have this addressed shortly. Apologies for the delay.

**Comment by benjaminkohl** [comment]
We just updated from 1.3.3 straight to 1.5.0 (locally) and now the export works for non-admin BUT using the "Run Report" tool results in an error because the `sprout_settings` table doesn't exist.

**Comment by BenParizek** [comment]
@benjaminkohl This should be resolved in Sprout Reports v1.5.1

**Comment by BenParizek** [comment]
@benjaminkohl @ryanmasuga I've wanted to follow up here as I just learned a bit more about the error you were running into, specifically the "sprout_settings table doesn't exist" error.

There was a bug in a recent migration that could reset all shared Sprout settings. This affected Sprout Reports as reports use a common codebase for reporting between Sprout Reports and Sprout Forms. The only way to fix the migration error is to roll-back and re-run the updates or retrieve your old settings from a backup database.

If you need to review the shared Sprout settings in the db directly, we are currently using a different method to save settings that Craft does with the Project Config. The Sprout Settings would be in the `sprout_settings` table and the below SQL statement reflects what those settings should roughly look like for Sprout Reports:

```
INSERT INTO `craft_sprout_settings` (`id`, `model`, `settings`, `dateCreated`, `dateUpdated`, `uid`) VALUES
('1', 'barrelstrength\\sproutbasereports\\models\\Settings', '{\"pluginNameOverride\":\"\",\"defaultPageLength\":\"50\",\"defaultExportDelimiter\":\";\"}', '2019-08-31 20:08:13', '2020-05-16 16:19:34', 'baa4d16d-32e5-43d1-8135-278e14cbc2ea');
```

_Note: For reasons I won't go into here, we currently manage some settings in this non-standard way (as the default way Craft provides to manage settings doesn't solve all the needs for our particular plugin use-case with some shared features between plugins). I recognize that this abstraction has caused more problems than are ideal and we're currently exploring ways to update the way settings are handled in our plugins to find a better way to solve the shared settings use case and get things working in a more Craft-native way._
